mongodump/mongodump.go

```diff
@@ -157,6 +158,8 @@ func (dump *MongoDump) Init() error {
 	}
 
 	dump.manager = intents.NewIntentManager()
+
+	dump.isWiredTiger = db.IsWiredTiger(dump.SessionProvider.DB(dump.ToolOptions.Namespace.DB), dump.ToolOptions.Namespace.Collection)
```

Comment:
It looks like this only works if the DB & collection are specified.  If they aren't, we still need to detect wiredTiger for single-DB or all-DB dumps.  In the ticket, I mentioned "admin.system.version" as a general check.  That may have permissions issues, and IIRC, "local.startup_log" might be another fallback.  Either way, needs to be tested against Atlas Free Tier Proxy and ADL -- if none of those collections are readable, then assuming wiredTiger is true is probably the right thing to do, as neither Free Tier nor ADL have MMAPv1.

I suggest adding a new function to m-t-c for the general case, as that's what mongomirror will need as well.